transport/src/main/java/io/netty/channel/DefaultChannelPipeline.java

```diff
@@ -1240,10 +1221,11 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
         public void channelReadComplete(ChannelHandlerContext ctx) throws Exception { }
     }
 
-    static final class HeadContext extends AbstractChannelHandlerContext
-            implements ChannelOutboundHandler {
+    final class HeadContext extends AbstractChannelHandlerContext
```

Comment:
@nmittler nope as it needs to call `callHandlerAddedForAllHandlers` `destroy()` and others ;)